kms++/src/videomode.cpp

```diff
@@ -110,18 +110,111 @@ static char sync_to_char(SyncPolarity pol)
 	}
 }
 
+#define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]))
+
+template<typename T>
+std::string join(const T& values, const std::string& delim)
+{
+	std::ostringstream ss;
+	for (const auto& v : values) {
+		if (&v != &values[0])
+			ss << delim;
+		ss << v;
+	}
+	return ss.str();
+}
+
+static const char *mode_type_names[] = {
+	// the first 3 are deprecated so don't care about a short name
+	"builtin", // deprecated
+	"clock_c", // deprecated
+	"crtc_c", // deprecated
+	"P", // "preferred",
+	"default", // deprecated,
+	"U", // "userdef",
+	"D", // "driver",
+};
+
+static const char *mode_flag_names[] = {
+	// the first 5 flags are displayed elsewhere
+	NULL, // "phsync",
+	NULL, // "nhsync",
+	NULL, // "pvsync",
+	NULL, // "nvsync",
+	NULL, // "interlace",
+	"dblscan",
+	"csync",
+	"pcsync",
+	"ncsync",
+	"hskew",
+	"bcast", // deprecated
+	"pixmux", // deprecated
+	"2x", // "dblclk",
+	"clkdiv2",
+};
+
+static const char *mode_3d_names[] = {
+	NULL, "3dfp", "3dfa", "3dla", "3dsbs", "3dldepth", "3dgfx", "3dtab", "3dsbs",
+};
+
+static const char *mode_aspect_names[] = {
+	NULL, "4:3", "16:9", "64:27", "256:135",
+};
+
+const string mode_type_str(uint32_t inval) {
+	uint32_t val = inval;
+	vector<string> v;
+	for (size_t i = 0; i < ARRAY_SIZE(mode_type_names); i++) {
+		if (val & (1 << i)) {
+			v.push_back(mode_type_names[i]);
+			val &= ~ (1 << i);
+		}
+	}
+	if (val != 0)
+		return fmt::format("0x{:x}", inval);
```

Comment:
This is one way to handle this. But how about adding the val as a string to the vector? Then we'd get something like
"P|D|0x70" where 0x70 are the unhandled flags.